### PR TITLE
MAT-7796: Edit dialog layout

### PR DESCRIPTION
**Pull request by mcmcphillips**
## MADiE PR

Jira Ticket: [MAT-7796](https://jira.cms.gov/browse/MAT-7796)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is in to the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [x] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/CqlBuilderPanel/functionsSection/functions/Functions.tsx [hunk 1]
```diff
@@ -62,11 +68,20 @@ const Functions = ({
...
 
+  // edit dialog utilities
+  const [discardDialog, setDiscardDialog] = useState({
+    open: false,
+    operation: null,
+  });
+  const [editFunctionDialogOpen, setEditFunctionDialogOpen] =
+    useState<boolean>();
+  const [deleteDialogOpen, setDeleteDialogOpen] = useState<boolean>(false);
+  const [selectedFunction, setSelectedFunction] = useState<Lookup>();
```
Comment (sb-cecilialiu) [inline src/CqlBuilderPanel/functionsSection/functions/Functions.tsx]: FunctionLookup actually extends Lookup, do you think it's ok to use FunctionLookup instead? On Line 164 with setSelectedFunction(), table.getRow(row.cell.row.id).original has all attributes of FunctionLookup.

File: src/CqlBuilderPanel/functionsSection/FunctionsSection.tsx [hunk 2]
```diff
@@ -18,7 +15,10 @@ export interface FunctionProps {
   cql: string;
   isCQLUnchanged: boolean;
   functions?: FunctionLookup[];
+  resetCql?: Function;
+  cqlBuilderLookupTypes?: any;
```
Comment (sb-cecilialiu) [inline src/CqlBuilderPanel/functionsSection/FunctionsSection.tsx]: Do we need cqlBuilderLookupTypes?

Comment (mcmcphillips) [reply src/CqlBuilderPanel/functionsSection/FunctionsSection.tsx]: We appear to be using it in this file and passing it to consuming <FunctionBuilder and <Functions

There's probably a way to refactor it, but I avoided doing anything outside the scope of setting up the layout

Comment (mcmcphillips) [reply src/CqlBuilderPanel/functionsSection/FunctionsSection.tsx]: Nevermind this appears to be completely a typo one letter off of the actual variable

File: src/CqlBuilderPanel/functionsSection/functions/Functions.tsx (hunk 1 above)
Comment (mcmcphillips) [reply src/CqlBuilderPanel/functionsSection/functions/Functions.tsx]: Yes, that would be the better way to do it

**Review by jkotanchik-SB — APPROVED** [review]
